### PR TITLE
Add decimal input support for array functions

### LINKED ISSUE #5940: Add decimal type signatures for array functions
### Description

Array functions implemented as vector functions do not support Decimal types currently because of missing signatures. Opening this issue to get inputs from the Velox community regarding adding these signatures.
cc: @aditi-pandit @majetideepak @karteekmurthys

### PR DESCRIPTION
**Pull request by pramodsatya**
Resolves #5940

### CODE REVIEW

File: velox/functions/prestosql/ArraySum.cpp [hunk 1]
```diff
@@ -261,6 +261,12 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
                                 .argumentType(fmt::format("array({})", argType))
                                 .build());
   }
+  signatures.emplace_back(exec::FunctionSignatureBuilder()
+                              .integerVariable("precision")
+                              .integerVariable("scale")
+                              .returnType("double")
```
Comment (karteekmurthys) [inline velox/functions/prestosql/ArraySum.cpp:267]: What is the return type for array_sum for decimals in java?

```
exec::AggregateFunctionSignatureBuilder()
          .integerVariable("a_precision")
          .integerVariable("a_scale")
          .argumentType("DECIMAL(a_precision, a_scale)")
          .intermediateType("VARBINARY")
          .returnType("DECIMAL(38, a_scale)")
          .build(),
```

Comment (karteekmurthys) [reply velox/functions/prestosql/ArraySum.cpp:267]: Also, please add array sum tests as well.

Comment (pramodsatya) [reply velox/functions/prestosql/ArraySum.cpp:267]: Sure, thanks for the suggestion! The return type is double for decimals: `https://prestodb.io/docs/current/functions/array.html#array_sum`

Comment (karteekmurthys) [reply velox/functions/prestosql/ArraySum.cpp:267]: Can you confirm from the presto-cli?

Comment (pramodsatya) [reply velox/functions/prestosql/ArraySum.cpp:277]: Sure, the return type is double from the query plan:
```
presto:tpch> explain analyze select array_sum(ARRAY[DECIMAL '1.2', DECIMAL '2.56']);
                                                                                             Query Plan                                       >
---------------------------------------------------------------------------------------------------------------------------------------------->
 Fragment 1 [SINGLE]                                                                                                                          >
     CPU: 12.60ms, Scheduled: 46.53ms, Input: 1 row (0B); per task: avg.: 1.00 std.dev.: 0.00, Output: 1 row (9B)                             >
     Output layout: [array_sum]                                                                                                               >
     Output partitioning: SINGLE []                                                                                                           >
     Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                            >
     - Project[projectLocality = LOCAL] => [array_sum:double]                                                                                 >
             CPU: 10.00ms (100.00%), Scheduled: 26.00ms (92.86%), Output: 1 row (9B)                                                          >
             Input avg.: 0.06 rows, Input std.dev.: 387.30%                                                                                   >
             array_sum := reduce([Block: position count: 2; size: 72 bytes], DOUBLE'0.0', (s$lambda, x$lambda) -> (s$lambda) + (COALESCE(x$lam>
         - LocalExchange[ROUND_ROBIN] () => []                                                                                                >
                 CPU: 0.00ns (0.00%), Scheduled: 2.00ms (7.14%), Output: 1 row (0B)                                                           >
                 Input avg.: 1.00 rows, Input std.dev.: 0.00%                                                                                 >
             - Values => []                                                                                                                   >
                     CPU: 0.00ns (0.00%), Scheduled: 0.00ns (0.00%), Output: 1 row (0B)                                                       >
                     Input avg.: 1.00 rows, Input std.dev.: 0.00%                                                                             >
                     ()
```

**Pull request closed without merging**